lib/workers/archive_conversations.ex

```diff
@@ -0,0 +1,11 @@
+defmodule ChatApi.Workers.ArchiveConversations do
+  use Oban.Worker, queue: :default
+
+  alias ChatApi.Conversations
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{} = _job) do
+    Conversations.list_conversations_to_archive()
+    |> Enum.each(&Conversations.archive_conversation/1)
```

Comment:
hmm i wonder if we should add a method to bulk-update/archive conversations, i.e. `Conversations.archive_conversations(conversations)`?

should we be using `update_all`, or `Ecto.Multi`, or something like that? i'm honestly not sure... (it might not matter too much in this case?)

Reply:
thank  you for the hint @reichert621 

I created the `archive_conversations/1` function and feel that is much better
I also added a Logger.info to the Oban Worker that logs the number of archived conversations
The test should have all the cases we need, i think

```elixir
  def archive_conversations(days_closed \\ 14) do
    Conversation
    |> where([c], is_nil(c.archived_at))
    |> where(status: "closed")
    |> where([c], c.updated_at < ago(^days_closed, "day"))
    |> Repo.update_all(set: [archived_at: DateTime.utc_now()])
  end
```

Reply:
nice! :)